### PR TITLE
[web-wallet] Add generate new keys option

### PR DESCRIPTION
**Pull request by kent-white**
## Motivation

#597 We want to allow users to generate new keys to use the wallet. We can do a lot better here with the UI, etc. But the base functionality is there with this. 

We also want to add api requests for the generation but I know there is another PR coming with networking so I'll get this in first.

## Test Plan

yarn start, yarn test

https://user-images.githubusercontent.com/19931667/165199610-e1cf1926-c047-4e5c-83c8-d13afe02caca.mov

## Related PRs

#593

### CODE REVIEW

File: ecosystem/web-wallet/src/pages/Login.tsx [hunk 1]
```diff
@@ -1,30 +1,56 @@
...
   const [, dispatch] = useGlobalState()
   const navigate = useNavigate()
 
   function handleSubmit (event: React.FormEvent<HTMLFormElement>) {
     event.preventDefault()
-    const encodedKey = Uint8Array.from(Buffer.from(key, 'hex'))
-    const account = new AptosAccount(encodedKey, undefined)
-    dispatch({ account })
-    navigate('/wallet')
+    const result = (loginAccount(key))
```
Comment (hariria) [inline ecosystem/web-wallet/src/pages/Login.tsx:18]: Is the outer parentheses necessary?

File: ecosystem/web-wallet/src/pages/Login.tsx [hunk 2]
```diff
@@ -1,30 +1,56 @@
...
 
   return (
     <div className="App-header">
       <h2>Aptos Wallet</h2>
       <form onSubmit={handleSubmit}>
-        <input onChange={(e) => setKey(e.target.value)}/>
+        <input onChange={onChange} value={key}/>
       </form>
+      <small className="Error-message">{error}</small>
+      <button onClick={onGenerateClick}>
```
Comment (hariria) [inline ecosystem/web-wallet/src/pages/Login.tsx:51]: Nit: indent

File: ecosystem/web-wallet/src/pages/Login.tsx [hunk 3]
```diff
@@ -1,30 +1,56 @@
...
   const navigate = useNavigate()
 
   function handleSubmit (event: React.FormEvent<HTMLFormElement>) {
     event.preventDefault()
-    const encodedKey = Uint8Array.from(Buffer.from(key, 'hex'))
-    const account = new AptosAccount(encodedKey, undefined)
-    dispatch({ account })
-    navigate('/wallet')
+    const result = (loginAccount(key))
+    if (result.isOk()) {
```
Comment (hariria) [inline ecosystem/web-wallet/src/pages/Login.tsx:19]: Haha this is very rusty

File: ecosystem/web-wallet/src/types.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  = Ok<T, E> // contains a success value of type T
+  | Err<T, E> // contains a failure value of type E
+
+/**
+ * Construct a new Ok result value.
+ */
+export const ok = <T, E>(value: T): Ok<T, E> => new Ok(value)
+
+/**
+  * Construct a new Err result value.
```
Comment (hariria) [inline ecosystem/web-wallet/src/types.tsx:39]: Nit indent

**Review by hariria — CHANGES_REQUESTED** [review]
Some nits

File: ecosystem/web-wallet/src/pages/Login.tsx (hunk 3 above)
Comment (kent-white) [reply ecosystem/web-wallet/src/pages/Login.tsx:19]: lol yea I felt weird doing it another way with try catches.

**Review by davidiw — COMMENTED** [review]
Awesome work!

Please add our license to the headers of these

**Comment by hariria** [comment]
LGTM!

**Pull request merged**